src/cli/install.ps1

```diff
@@ -1,176 +1,220 @@
+param(
+    [Parameter(ParameterSetName = "Release")]
+    [ValidateSet("Canary")] # TODO: Add "Latest" to the ValidateSet once Bun is stable.
+    $Release = "Canary", # TODO: change the default value to "Latest" once Bun is stable.
+
+    [Parameter(
+        ParameterSetName = "Version",
+        HelpMessage = "Invalid version format. Please use SemVer notation (e.g., 1.2.3)"
+    )]
+    [ValidatePattern("^(?<Major>\d+)\.(?<Minor>\d+)\.(?<Patch>\d+)$")]
+    $Version,
+
+    [switch]
+    $ForceBaseline,
+
+    [switch]
+    $InstallToUserProfile
+)
+
+#region Prologue
+if ($PSCmdlet.ParameterSetName -eq "Version") {
+    Write-Error "Parameter -Version is not yet supported." -ErrorAction Stop
+}
+
+Write-Warning "Bun for Windows is currently experimental.`nFor a more stable experience, please install Bun within WSL:`nhttps://bun.sh/docs/installation"
+
+trap {
+    if ($installationPath) {
+        Write-Verbose "Installation failed. Cleaning up temporary files, installation directory and PATH variable."
+        
+        Remove-Item -LiteralPath $installer
+        Remove-Item -LiteralPath $installationPath -Recurse
+        
+        [Environment]::SetEnvironmentVariable("BUN_HOME", $null, "User")
+        Remove-Item env:\BUN_HOME -ErrorAction Ignore
+        if ([Environment]::GetEnvironmentVariable("PATH", "User") -like "*%BUN_HOME%*") {
+            $oldUserPATH = [Environment]::GetEnvironmentVariable("PATH", "User")
+            $newUserPATH = $oldUserPATH.Replace("%BUN_HOME%", "").Replace(";;", ";")
+            [Environment]::SetEnvironmentVariable("PATH", $newUserPATH, "User")
+        }
+        $env:Path = [Environment]::ExpandEnvironmentVariables(
+            [Environment]::GetEnvironmentVariable("Path", "Machine") + 
+            ";" + 
+            [Environment]::GetEnvironmentVariable("Path", "User")
+        )
+    }    
+    break
+}
+
+#region Requirements Check
+# Ensure this script is run on Windows and meets the minimum Windows version requirement
+if ($env:OS -eq "Windows_NT") {
+    $WinVer = [System.Environment]::OSVersion.Version
+    if ($WinVer.Build -lt 17763) {
+        Write-Error "Bun requires Windows 10 version 1809 or newer." -ErrorAction Stop
+    }
+} else {
+    Write-Error "This script must be run on Windows." -ErrorAction Stop
+}
+
+# Check for AMD64 support
+if ($env:PROCESSOR_ARCHITECTURE -ne "AMD64") {
+    Write-Error "Bun for Windows is only available for Windows x64." -ErrorAction Stop
+}
+
+# Check for AVX2 support
+$MethodDefinition = @"
+[DllImport("kernel32.dll")]
+public static extern long GetEnabledXStateFeatures();
+"@
+$kernel32 = Add-Type -MemberDefinition $MethodDefinition -Name kernel32 -Namespace Win32 -PassThru -ErrorAction Stop
+$avx2Support = [bool]($kernel32::GetEnabledXStateFeatures() -band 0b0100)
+if ($avx2Support -and $ForceBaseline) {
+    # We should protect the user from installing a variant with limited functionality, although it is not needed.
```

Comment:
"limited functionality" is incorrect. this binary is just slower because it cannot take advantage of the AVX2 instructions that this binary does not ship with.